rootfs/etc/nginx/lua/configuration.lua

```diff
@@ -27,13 +32,75 @@ local function fetch_request_body()
   return body
 end
 
+-- Returns the full chain certificate for a given host
+function _M.get_cert(host_name)
+  return configuration_data:get(host_name .. cert_ext)
+end
+
+-- Returns the private key for a given host
+function _M.get_cert_key(host_name)
+  return configuration_data:get(host_name .. key_ext)
+end
+
+-- Handler for the /configuration/certificates endpoint
+local function handle_cert_request()
+  if ngx.var.request_method ~= "POST" then
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    ngx.print("Only POST requests are allowed!")
+    return
+  end
+
+  local cert_str = fetch_request_body()
+
+  local ok, certs = pcall(json.decode, cert_str)
+  if not ok then
+    ngx.log(ngx.ERR,  "could not parse certificate: " .. tostring(certs))
+    return
+  end
+
+  if not certs then
+    ngx.log(ngx.ERR, "certificate dynamic-configuration: unable to read valid request body")
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    return
+  end
+
+  -- Update full chain certificates and private keys for each host
+  for _, cert in pairs(certs) do
+    if cert.HostName and cert.Cert and cert.FullChainCert then
+      if cert.FullChainCert == "" then
+        cert.FullChainCert = cert.Cert
+      end
+      local success, err = configuration_data:set(cert.HostName .. cert_ext, cert.FullChainCert)
+      if not success then
+        ngx.log(ngx.ERR, "certificate dynamic-configuration: error setting certificate: " .. tostring(err))
```

Comment:
Could we dump cert.Hostname here as well? Or is that composite in the `err` that would be thrown here?